tools/get-keys-from-ts.js

```diff
@@ -0,0 +1,451 @@
+        return false;
+    }
+
+    if (annotationType.type === "TSTupleType") {
+        return annotationType.elementTypes.some(annType => checkTraversability(annType));
+    }
+
+    if (notTraversableTypes.has(annotationType.typeName.name)) {
+        return false;
+    }
+
+    if (!isTraversable(annotationType.typeName.name)) {
+
+        // Todo?
+        /*
+        const innerInterfaceName = tsAnnotation.typeName.name;
+        const innerTsDeclarationNode = findTsInterfaceDeclaration(innerInterfaceName);
+
+        if (!innerTsDeclarationNode) {
+
+            const innerTsTypeNode = findTsTypeDeclaration(innerInterfaceName);
+
+            // We might iterate types here to see if children are iterable and
+            //   fail if not
+
+            unrecognizedTSTypeReferences.add(`${tsAnnotation.typeName.name}`);
+            break;
+        }
+
+        // We might iterate interfaces here to see if children are iterable
+        //   (see `addNodeForInterface` for a pattern of iteration)
+        */
+
+        throw new Error(`Type unknown as to traversability: ${annotationType.typeName.name}`);
+    }
+
+    return true;
+}
+
+/**
+ * Get the literal names out of AST
+ * @param {Node} excludedItem Excluded node
+ * @returns {string[]} The literal names
+ */
+function findOmitTypes(excludedItem) {
+    if (excludedItem.type === "TSUnionType") {
+        return excludedItem.types.map(typeNode => findOmitTypes(typeNode));
+    }
+    if (excludedItem.type !== "TSLiteralType") {
+        throw new Error("Processing of non-literals in `Omit` not currently supported");
+    }
+    return excludedItem.literal.value;
+}
+
+/**
+ * Checks whether property should be excluded
+ * @param {string} property Property to check
+ * @param {string[]} excludedProperties Properties not to allow
+ * @returns {boolean} Whether or not to be excluded
+ */
+function isPropertyExcluded(property, excludedProperties) {
+    return propertiesToIgnore.has(property) ||
+        (excludedProperties && excludedProperties.includes(property));
+}
+
+//------------------------------------------------------------------------------
+// Public APIs
+//------------------------------------------------------------------------------
+
+/**
+ * Returns alphabetized keys
+ * @param {KeysStrict} initialNodes Initial node list to sort
+ * @returns {KeysStrict} The keys
+ */
+function alphabetizeKeyInterfaces(initialNodes) {
+
+    /**
+     * Alphabetize
+     * @param {string} typeA The first type to compare
+     * @param {string} typeB The second type to compare
+     * @returns {1|-1} The sorting index
+     */
+    function alphabetize([typeA], [typeB]) {
+        return typeA < typeB ? -1 : 1;
+    }
+    const sortedNodeEntries = Object.entries(initialNodes).sort(alphabetize);
+
+    /**
+     * Get the key sorter for a given type
+     * @param {string} type The type
+     * @returns {(string, string) => -1|1} The sorter
+     */
+    function getKeySorter(type) {
```

Comment:
Would it make sense to get the order of keys from `lib/visitor-keys.js` for known nodes, and just add new nodes (and new keys for known nodes) with any order of keys? That would avoid updating this logic whenever there's something new that isn't covered by it.

Reply:
I followed this approach but making it just a little more transparently deterministic, I added a commit to alphabetize the remainder at the end of any existing keys.